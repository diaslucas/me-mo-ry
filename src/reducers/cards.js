import { FLIP_CARD } from '../constants';
import cardList from '../cardList';

export default (state = cardList, action) => {
  switch (action.type) {
    case FLIP_CARD: {
      const index = action.payload;
      return [
        ...state.slice(0, index),
        { ...state[index], isFlipped: !state[index].isFlipped },
        ...state.slice(index + 1),
      ];

    }

    default:
      return state;
  }
};
