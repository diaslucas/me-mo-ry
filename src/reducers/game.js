import { SET_LAST_CARD_FLIPPED, SET_CURRENT_PLAYER, ADD_POINT, TOGGLE_SOUND } from '../constants';
import { numberOfPairs } from '../cardList';

const initialState = {
  isOver: false,
  currentPlayer: 1,
  cardsGuessed: 0,
  lastCardFlipped: { card: null, index: null },
  isSoundOn: true,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_LAST_CARD_FLIPPED: {
      const { card, index } = action.payload;
      return {
        ...state,
        lastCardFlipped: { card, index },
      };
    }

    case SET_CURRENT_PLAYER: {
      return {
        ...state,
        currentPlayer: action.payload,
      };
    }

    case ADD_POINT: {
      const cardsGuessed = state.cardsGuessed + 1;
      const isOver = cardsGuessed === numberOfPairs;
      return {
        ...state,
        isOver,
        cardsGuessed,
      };
    }

    case TOGGLE_SOUND: {
      const isSoundOn = !state.isSoundOn;
      return {
        ...state,
        isSoundOn,
      };
    }

    default:
      return state;
  }
};
