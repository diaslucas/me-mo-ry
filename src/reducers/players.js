import { ADD_POINT, SET_PLAYER_NAME } from '../constants';

const initialState = { 1: { id: 1, name: 'Player 1', points: 0 }, 2: { id: 2, name: 'Player 2', points: 0 } };

export default (state = initialState, action) => {
  switch (action.type) {
    case ADD_POINT: {
      const playerId = action.payload;
      const playerState = state[playerId];
      const { points } = state[playerId];
      return {
        ...state,
        [playerId]: {
          ...playerState,
          points: points + 1,
        },
      };
    }

    case SET_PLAYER_NAME: {
      const { playerId, name } = action.payload;
      const playerState = state[playerId];
      return {
        ...state,
        [playerId]: {
          ...playerState,
          name,
        },
      };
    }

    default:
      return state;
  }
};
