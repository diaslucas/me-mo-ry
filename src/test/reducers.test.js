/* eslint-disable no-undef */
import cards from '../reducers/cards';
import game from '../reducers/game';
import players from '../reducers/players';


describe('Testing reducers', () => {
  describe('Cards Reducer', () => {
    test('Initial state match snapshot', () => {
      expect(cards).toMatchSnapshot();
    });
  });

  describe('Game Reducer', () => {
    test('Initial state match snapshot', () => {
      expect(game).toMatchSnapshot();
    });
  });

  describe('Players Reducer', () => {
    test('Initial state match snapshot', () => {
      expect(players).toMatchSnapshot();
    });
  });
});
