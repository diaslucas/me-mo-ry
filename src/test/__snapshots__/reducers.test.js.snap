// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing reducers Cards Reducer Initial state match snapshot 1`] = `[Function]`;

exports[`Testing reducers Game Reducer Initial state match snapshot 1`] = `[Function]`;

exports[`Testing reducers Players Reducer Initial state match snapshot 1`] = `[Function]`;
